contracts/upgradeable_contracts/erc20_to_native/HomeBridgeErcToNative.sol

```diff
@@ -29,8 +29,12 @@ contract HomeBridgeErcToNative is EternalStorage, BasicBridge, BasicHomeBridge,
             uint256 fee = calculateFee(valueToTransfer, false, feeManager, HOME_FEE);
             valueToTransfer = valueToTransfer.sub(fee);
         }
-        setTotalBurntCoins(totalBurntCoins().add(valueToTransfer));
-        address(0).transfer(valueToTransfer);
+        uint256 valueToBurn = valueToTransfer;
+        if(isPOSDAOFeeManager()) {
```

Comment:
It looks like this method should be called only in case if the fee manager is configured. So, could you consider the following changes:
- introduce the new method in the POSDAO fee manager: `getAmountToBurn`. This will allow to locate all logic related to fee handling in the fee manager.
- change the fall back method in the bridge contract as following:
```solidity
// . . .
// . . .
uint256 valueToTransfer = msg.value;
uint256 valueToBurn = valueToTransfer;
address feeManager = feeManagerContract();
if (feeManager != address(0)) {
    valueToBurn = getAmountToBurn(valueToTransfer)
    uint256 fee = calculateFee(valueToTransfer, false, feeManager, HOME_FEE);
    valueToTransfer = valueToTransfer.sub(fee);
}
setTotalBurntCoins(totalBurntCoins().add(valueToBurn));
address(0).transfer(valueToBurn);
emit UserRequestForSignature(msg.sender, valueToTransfer);
```

Reply:
Thanks for the suggestions. Please check https://github.com/poanetwork/poa-bridge-contracts/pull/164/commits/e605559f1727ef8dac6d75a27e2d05bba87043f3